app/controllers.py

```diff
@@ -1,31 +1,44 @@
-from werkzeug.security import generate_password_hash, check_password_hash
-from flask import jsonify, json
 import datetime
+import os
+
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask import jsonify
 import jwt
+from sqlalchemy import or_
 
 from app import db, app
 from app.models import User, Budget, MiniBudget
 from app.schemas import UserSchema, BudgetSchema, MiniBudgetSchema
 
+
 class UserController():
     def register(self, username, email, password1, password2):
         if password1 != password2:
             return "Passwords do not match"
-        user = User.query.filter_by(username = username, email = email).first()
+        user = User.query.filter(
+            or_(User.username == username, User.email == email)
```

Comment:
Using SQLAlchemy's `or_` here makes certain that neither a username or email with the same value exists. The previous query checked if there was a row with the same email and username but wouldn't catch rows that have the same username but different email etc...